samples/BuilderApp/Program.cs

```diff
@@ -101,6 +101,8 @@ private static void Usage()
         }
     }
 
+    // Class for performing the project build
+    // The Microsoft.Build dlls needed from within aren't loaded until after the class is instantiated
```

Comment:
It's actually more fine-grained. Assemblies get loaded just in time to call the first _method_ that uses them. I tried to find documentation on this, but didn't find anything official. [This StackOverflow answer](https://stackoverflow.com/a/2967279) quotes a book explanation:

> In the CLR, loading typically is triggered by the just in time (JIT) compiler based on types. When the JIT compiler tries to convert a method body from CIL to machine code, it needs access to the type definition of the declaring type as well as the type definitions for the type's fields.

Can you update this to something like

> The Microsoft.Build namespaces must be referenced from a method that is called after RegisterInstance so that it has a chance to change their load behavior. Here, we put them into a separate class.